cudax/include/cuda/experimental/__execution/write_env.cuh

```diff
@@ -114,10 +163,35 @@ _CCCL_TRIVIAL_API constexpr auto write_env_t::operator()(_Sndr __sndr, _Env __en
   return __sndr_t<_Sndr, _Env>{{}, static_cast<_Env&&>(__env), static_cast<_Sndr&&>(__sndr)};
 }
 
+template <class _Env>
+_CCCL_TRIVIAL_API constexpr auto write_env_t::operator()(_Env __env) const
```

Comment:
Nitpick: those should be ``[[nodiscard]]`

Any reason to move them out of the class definition?